Content.Shared/RCD/RCDPrototype.cs

```diff
@@ -0,0 +1,114 @@
+using Content.Shared.Physics;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.RCD;
+
+/// <summary>
+/// Contains the parameters for a RCD construction / operation
+/// </summary>
+[Prototype("rcd")]
+public sealed class RCDPrototype : IPrototype
+{
+    [IdDataField]
+    public string ID { get; private set; } = default!;
+
+    /// <summary>
+    /// The RCD mode associated with the operation
+    /// </summary>
+    [DataField(required: true), ViewVariables(VVAccess.ReadOnly)]
+    public RcdMode Mode { get; private set; } = RcdMode.Invalid;
+
+    /// <summary>
+    /// The name associated with the prototype
+    /// </summary>
+    [DataField("name"), ViewVariables(VVAccess.ReadOnly)]
+    public string SetName { get; private set; } = "Unknown";
+
+    /// <summary>
+    /// The name of the radial container that this prototype will be listed under on the RCD menu
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public string Category { get; private set; } = "Undefined";
+
+    /// <summary>
+    /// Texture path for this prototypes menu icon
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public SpriteSpecifier? Sprite { get; private set; } = null;
+
+    /// <summary>
+    /// The entity prototype that will be constructed (mode dependent)
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public string? Prototype { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// Number of charges consumed when the operation is completed
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public int Cost { get; private set; } = 1;
+
+    /// <summary>
+    /// The length of the operation 
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadOnly)]
+    public float Delay { get; private set; } = 1f;
+
+    /// <summary>
+    /// The visual effect that plays during this operation
+    /// </summary>
+    [DataField("fx"), ViewVariables(VVAccess.ReadOnly)]
+    public EntProtoId? Effect { get; private set; } = null;
+
+    /// <summary>
+    /// A list of rules that govern where the entity prototype can be contructed
+    /// </summary>
+    [DataField("rules"), ViewVariables(VVAccess.ReadOnly)]
+    public HashSet<RcdConstructionRule> ConstructionRules { get; private set; } = new();
+
```

Comment:
Would need to check but RulesPrototype probably covers this though this is fine for now if we don't add much more.

Reply:
I can look into this later I think, it looks like RulesSystem would need a clean-up beforehand